index.js

```diff
@@ -342,6 +342,25 @@ module.exports = {
     },
 
     /**
+     * Call this if you plan on loading TypeScript files.
+     *
+     * Encore.configureTypeScript(function(tsConfig) {
+     *      // change the tsConfig
+     * });
+     *
+     * Supported configuration options:
+     * @see https://github.com/TypeStrong/ts-loader/blob/master/README.md#available-options
+     *
+     * @param {function} callback
+     * @return {exports}
+     */
+    configureTypeScript(callback) {
+        webpackConfig.configureTypeScript(callback);
```

Comment:
Don't worry about that, the idea is to get to the best solution possible.

Regarding why I did accept your callback suggestion before. I indeed thought the same as you, on seconds thoughts, after looking the code again, Is it better to keep it like options!?

My conclusion was: no. Why? Because I think it is more explicit to have a callback which accept parameters to configure instead having to know the structure of an object configuration that configure several aspects of application. for example.

This is how i thought on how could be integrated other stuff  better in future:
* `fork-ts-checker-webpack-plugin`:  When tried to add this in `ts-loader` configuration did not feel natural since it is another plugin that can be configured by its own. Still is only useful if you use `ts-loader` and with the option: `transpileOnly: true`. In future this could be passed as another config object to callback.
* I also thought on the `happypack` plugin. This one is more general and can be used with many loaders, but still, having the call back would help at the time of configuration because you will have to configure a bit different the loaders when using it.

Because of this I did not question the callback idea. And that's why I think it would be better like this. Maybe thinking about adding parameters to a configuration callback could not be an optimal solution either, in which case I am open to suggestions :)

What do you think about it?